src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,161 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> tempNode = new Node<>(null, value, null);
+        if (size == 0) {
+            head = tempNode;
+        } else {
+            tail.next = tempNode;
+            tempNode.prev = tail;
+        }
+        tail = tempNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == 0) {
+            if (size == 0) {
+                Node<T> current = new Node<>(null, value, null);
+                head = current;
+                tail = current;
+            } else {
+                Node<T> newHead = new Node<>(null, value, head);
+                newHead.next = head;
+                head.prev = newHead;
+                head = newHead;
+            }
+        } else if (index == size) {
+            Node<T> newTail = new Node<>(tail, value, null);
+            newTail.prev = tail;
+            tail.next = newTail;
+            tail = newTail;
+        } else {
+            Node<T> nodeByIndex = getNode(index);
+            Node<T> middleCurrent = new Node<>(nodeByIndex.prev, value, nodeByIndex);
+            nodeByIndex.prev.next = middleCurrent;
+            middleCurrent.prev = nodeByIndex.prev;
+            nodeByIndex.prev = middleCurrent;
+            middleCurrent.next = nodeByIndex;
+        }
+        size++;
```

Comment:
Your implementation of the add method works correctly, but it seems a bit more complex than necessary. Let's review and see how we could simplify it.

Redundant assignments: There are several instances where you are doing the same assignment twice. For example, newHead.next = head; and newTail.prev = tail; are redundant because you are already setting these fields in the Node constructor.

Handling head and tail separately: Your code specifically handles the cases where the index is 0 (head) or equals to size (tail). While this does make sense, it results in duplicated logic that can be avoided. With a proper check for the previous node being null, the "head" case can be handled together with the general case, thus simplifying the code. The "tail" case can be handled as an append operation.

Prev and next references: The references to the previous and next nodes are somewhat scattered throughout your code. Grouping them together can make the logic more clear.

Here is a simpler algorithm to follow:

* If the index equals to size, treat it as an append operation and call the add(value) method directly.
Check the index to ensure it's valid.
* Retrieve the node currently at the specified index. This will become the "next" node for the new node.
* Create a new node with its next reference pointing to the retrieved node and its prev reference pointing to the prev reference of the retrieved node.
* If the prev reference of the new node is null, it means we are inserting at the head, so update the first reference to point to the new node. Otherwise, update the next reference of the prev node to point to the new node.
* Finally, update the prev reference of the next node (the node retrieved initially) to point to the new node.
* Remember to increment the size.

This way, your code will be more concise and to the point. I hope these suggestions are helpful and looking forward to your updated contribution.